### DISCUSSION TITLE
Understanding how Context Propagation is used

### DISCUSSION THREAD

**Discussion by moonbox3** [open-telemetry/opentelemetry-dotnet#5076] [Q&A]
I came across the following concept in OpenTelemetry and was wondering if it can be used to help us: https://opentelemetry.io/docs/concepts/signals/traces/#context-propagation. 

Here is our use-case:

We are exploring the use of the OTel dotnet SDK and need a method to propagate information from nested function calls back to the original caller. Specifically, if we implement the SDK, and one component calls another component that generates a metric we wish to track, can we use context propagation to add this metric (potentially as a key-value pair) to the calling function's context? This would enable the caller to access this metric when the call stack unwinds.

Thank you for your help.

**Comment by vishweshbankwar** [comment]
Context propagation only works downstream. For e.g. A calls B, you can pass information from A to B. You can use Baggage for this 
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api#baggage-api

In your case, looks like you want to pass information from B to A? (not sure if I follow your use case completely). 

Why cannot you track the metric directly using https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics?
